### ISSUE TITLE
Custom modal template overriding placement classes.

### ISSUE THREAD

**Issue by Thinkscape** [mgcrea/angular-strap#479]
Example, `template="foo.html"` where foo.html is something like:

``` html
<div class="modal {{type}}" tabindex="-1" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header" ng-hide="!title">
        <button type="button" class="close" ng-click="$close(false)" aria-hidden="true">&times;</button>
        <h4 class="modal-title">{{ title }}</h4>
      </div>
      <div class="modal-body">
      </div>
    </div>
  </div>
</div>
```

Notice that there's a `{{type}}` interpolation in the class name of the root element. Now we'll create a new modal with:

``` js
$modal({ placement: 'center', template: 'foo.html', show: true });
```

This will produce a modal **without** placement class (given that `type="confirm"` in this example):
![image](https://f.cloud.github.com/assets/270528/2169444/1d866004-955a-11e3-8923-c5d43914416c.png)

Here's the root of the problem.

Classes are assigned to the main element **before** the $digest is invoked (in which the template might define its own clasess)
![image](https://f.cloud.github.com/assets/270528/2169414/b69df23a-9559-11e3-8a01-8b688a803323.png)

So basically at line 131 all those classes are erased as angular is transcluding `class` attribute.
